src/main/scala/it/pagopa/interop/partyprocess/api/impl/PublicApiServiceImpl.scala

```diff
@@ -68,9 +68,7 @@ class PublicApiServiceImpl(
         partyManagementService.getInstitutionId(legalUser.relationshipId)
       )
       institutionInternalId = istitutionId.map(_.to.toString)
-      legalEmails           = institutionInternalId.map(id =>
-        legalUsers.filter(_.email.get(id).nonEmpty).head.email.get(id).orNull
-      )
+      legalEmails = institutionInternalId.map(id => legalUsers.filter(_.email.get(id).nonEmpty).head.email.get(id))
```

Comment:
Making some assumptions of your types:
```suggestion
      legalEmails = institutionInternalId.flatMap(id => legalUsers.flatMap(_.email.get(id)).headOption)
```

FYI `head` throws an error if the list is empty

Reply:
I've done a refactor